.github/workflows/hackage.yml

```diff
@@ -0,0 +1,127 @@
+        ghc: ["8.10.6", "8.8.4", "8.6.5"]
+
+    steps:
+
+      - uses: actions/checkout@v2
+        with:
+          submodules: true
+
+      - uses: haskell/actions/setup@v1
+        with:
+          ghc-version: ${{ matrix.ghc }}
+          cabal-version: '3.4'
+
+      - name: Cache Cabal
+        uses: actions/cache@v2
+        env:
+          cache-name: cache-cabal
+        with:
+          path: |
+            ~/.cabal/packages
+            ~/.cabal/store
+          key: v2-${{ runner.os }}-${{ matrix.ghc }}-hackage-${{ hashFiles('**/*.cabal') }}
+          restore-keys: |
+            v2-${{ runner.os }}-${{ matrix.ghc }}-hackage-
+            v2-${{ runner.os }}-${{ matrix.ghc }}-build-
+            v2-${{ runner.os }}-${{ matrix.ghc }}
+
+      - name: "Run cabal check"
+        run: |
+          if [[ ${{ matrix.package }} == *plugin ]]; then
+            cd plugins
+          fi
+          if [[ ${{ matrix.package }} != haskell-language-server ]]; then
+            cd ${{ matrix.package }}
+          fi
+          cabal check
+
+      - name: "Generate package dist tarball"
+        id: generate-dist-tarball
+        run: |
+          if [[ ${{ matrix.package }} == haskell-language-server ]]; then
+            cabal sdist --builddir=./
+          else
+            cabal sdist ${{ matrix.package }} --builddir=./
+          fi
+          echo ::set-output name=path::$(ls ./sdist/${{ matrix.package }}-*)
+
+      - name: "Unpack package source in an isolated location"
+        run: cabal unpack ${{ steps.generate-dist-tarball.outputs.path }} --destdir=./incoming
+
+      - name: "Try to get the current hackage version"
+        id: get-hackage-version
+        run: |
+          cd ./incoming
+          if cabal get $(ls -d ${{ matrix.package }}-*) --destdir=../current; then
+            echo ::set-output name=exists::true
+          fi
+
+      - name: "Compare the incoming and the current hackage version of the package"
+        id: compare-current-version
+        if: steps.get-hackage-version.exists == 'true'
+        run: |
+          # This will throw an error if there is any difference casue we have to bump up the package version
+          diff -qr -x "*.md" -x "data" $(ls -d ./incoming/${{ matrix.package }}-*) $(ls -d ./current/${{ matrix.package }}-*)
+
+      - name: "Create appropiate cabal.project"
+        if: steps.get-hackage-version.exists != 'true'
+        run: |
+          cd $(ls -d ./incoming/${{ matrix.package }}-*)
+          echo "packages: . ../../* ../../plugins/*" > cabal.project
+          # TODO: remove when not needed
+          echo "allow-newer: Chart-diagrams:diagrams-core, SVGFonts:diagrams-core" >> cabal.project
+
+      - name: "Build all package components in isolation"
+        if: steps.get-hackage-version.exists != 'true'
+        run: |
+          cd $(ls -d ./incoming/${{ matrix.package }}-*)
+          cabal build --enable-tests --enable-benchmarks
```

Comment:
```suggestion
          cabal build --enable-tests
```

Reply:
ideally benchmarks should be able to run using the hackage version